File: rb/features.rb
# GlobalSharkAttack SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module GlobalSharkAttackFeatures
  def self.make_feature(name)
    case name
    when "base"
      GlobalSharkAttackBaseFeature.new
    when "ratelimit"
      GlobalSharkAttackRatelimitFeature.new
    when "retry"
      GlobalSharkAttackRetryFeature.new
    when "test"
      GlobalSharkAttackTestFeature.new
    when "timeout"
      GlobalSharkAttackTimeoutFeature.new
    else
      GlobalSharkAttackBaseFeature.new
    end
  end
end

File: py/features.py
# GlobalSharkAttack SDK feature factory

from feature.base_feature import GlobalSharkAttackBaseFeature
from feature.test_feature import GlobalSharkAttackTestFeature


def _make_feature(name):
    features = {
        "base": lambda: GlobalSharkAttackBaseFeature(),
        "test": lambda: GlobalSharkAttackTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
